File: opcua-webservice-spring/core/src/main/java/org/opcfoundation/webserver/addressspace/nodes/UaObject.java
package org.opcfoundation.webserver.addressspace.nodes;

import org.eclipse.milo.opcua.stack.core.AttributeId;
import org.eclipse.milo.opcua.stack.core.NodeIds;
import org.eclipse.milo.opcua.stack.core.types.builtin.LocalizedText;
import org.eclipse.milo.opcua.stack.core.types.builtin.NodeId;
import org.eclipse.milo.opcua.stack.core.types.builtin.Variant;
import org.eclipse.milo.opcua.stack.core.types.builtin.unsigned.UByte;
import org.eclipse.milo.opcua.stack.core.types.enumerated.BrowseDirection;
import org.eclipse.milo.opcua.stack.core.types.enumerated.NodeClass;
import org.springframework.lang.Nullable;
import org.opcfoundation.webserver.addressspace.nodes.builtin.UaReferenceTypes;

import java.util.List;

public class UaObject extends UaInstanceNode {
    private final UaObjectType typeDefinition;

    public UaObject(
            NodeId nodeId,
            String browseName,
            LocalizedText displayName,
            UaObjectType typeDefinition)
    {
        super(nodeId, browseName, displayName);

        this.typeDefinition = typeDefinition;
        this.addReference(new UaReference(typeDefinition, UaReferenceTypes.HasTypeDefinition, true));
    }

    @Override
    public NodeClass nodeClass() {
        return NodeClass.Object;
    }

    public UaObjectType typeDefinition() {
        return typeDefinition;
    }

    public @Nullable UaNode parentNode()
    {
        List<UaReference> referenceList = getReferences(BrowseDirection.Inverse);

        for (UaReference item: referenceList)
        {
            if (item.reference().nodeId().equals(NodeIds.HasComponent))
            {
                return item.linkedNode();
            };
        }

        return null;
    }

    public void addMember(UaInstanceNode member)
    {
        addMemberNode(member);
    }

    public void organizes(UaNode node)
    {
        this.addReference(new UaReference(node, UaReferenceTypes.Organizes, true));
        node.addReference(new UaReference(this,UaReferenceTypes.Organizes, false));
    }

    @Override
    public Variant getAttribute(int attributeId)
    {
        if (AttributeId.EventNotifier.id() == attributeId)
        {
            return new Variant(UByte.valueOf(0));
        }

        return super.getAttribute(attributeId);
    }
}

File: opcua-webservice-node/src/server/addressspace/nodes/UaObject.ts
import { Attributes, BrowseDirection, NodeClass } from "opcua-webapi";
import { UaLocalizedText, UaNodeId, UaVariant, UaVariantType } from "opcua-webapi-ts";
import { UaInstanceNode } from "./UaInstanceNode";
import { UaNode } from "./UaNode";
import { UaReference } from "./UaReference";
import { UaReferenceTypes } from "./builtin/UaReferenceTypes";
import type { UaObjectType } from "./UaObjectType";

export class UaObject extends UaInstanceNode {
    private readonly _typeDefinition: UaObjectType;

    constructor(
        nodeId: UaNodeId,
        browseName: string,
        displayName: UaLocalizedText,
        typeDefinition: UaObjectType) {
        super(nodeId, browseName, displayName);
        this._typeDefinition = typeDefinition;
        this.addReference(new UaReference(typeDefinition, UaReferenceTypes.HasTypeDefinition, true));
    }

    get nodeClass(): NodeClass {
        return NodeClass.Object;
    }

    get typeDefinition(): UaObjectType {
        return this._typeDefinition;
    }

    get parentNode(): UaNode | null {
        const referenceList = this.getReferences(BrowseDirection.Inverse);

        for (const item of referenceList) {
            if (item.reference.nodeId.equal(UaReferenceTypes.HasComponent.nodeId)) {
                return item.linkedNode;
            }
        }

        return null;
    }

    addMember(member: UaInstanceNode): void {
        this.addMemberNode(member);
    }

    organizes(node: UaNode): void {
        this.addReference(new UaReference(node, UaReferenceTypes.Organizes, true));
        node.addReference(new UaReference(this, UaReferenceTypes.Organizes, false));
    }

    getAttribute(attributeId: number): UaVariant {
        if (Attributes.EventNotifier === attributeId) {
            return UaVariant.integer(0, UaVariantType.Byte);
        }

        return super.getAttribute(attributeId);
    }
}